### PR TITLE
Do not use Blob/FileReader modules in UWP

### LINKED ISSUE #10036: Invalid response for blob
### Problem Description

Testing the migration from .66 to .68 and we are seeing a fetch error "Invalid response for blob".

**Further**
Issue also exists on .67 but a fresh install at .66 is ok.

Error: Invalid response for blob: eyJlcnJvciI6IlVuYWJsZSB0byByZXNvbHZlIG1vZHVsZSAuL2RlYnVnZ2VyLXVpL2RlYnVnZ2VyV29ya2VyLmFjYTE3M2M0IGZyb20gQzpcXFVzZXJzXFxqYW1lc1xcRGVza3RvcFxcU2l0ZU1lc2gvLjogLi9kZWJ1Z2dlci11aS9kZWJ1Z2dlcldvcmtlci5hY2ExNzNjNCBjb3VsZCBub3QgYmUgZm91bmQgd2l0aGluIHRoZSBwcm9qZWN0IG9yIGluIHRoZXNlIGRpcmVjdG9yaWVzOlxuICBub2RlX21vZHVsZXMifQ==
    at XMLHttpRequest.get (index.bundle?platform=windows&dev=true&hot=false&inlineSourceMap=true:30545:21)
    at XMLHttpRequest.xhr.onload (index.bundle?platform=windows&dev=true&hot=false&inlineSourceMap=true:25151:46)
    at XMLHttpRequest.dispatchEvent (index.bundle?platform=windows&dev=true&hot=false&inlineSourceMap=true:31774:27)
    at XMLHttpRequest.setReadyState (index.bundle?platform=windows&dev=true&hot=false&inlineSourceMap=true:30894:18)
    at XMLHttpRequest.__didCompleteResponse (index.bundle?platform=windows&dev=true&hot=false&inlineSourceMap=true:30664:16)
    at index.bundle?platform=windows&dev=true&hot=false&inlineSourceMap=true:30811:47
    at EventEmitter.emit (index.bundle?platform=windows&dev=true&hot=false&inlineSourceMap=true:2162:37)
    at MessageQueue.__callFunction (index.bundle?platform=windows&dev=true&hot=false&inlineSourceMap=true:3128:31)
    at index.bundle?platform=windows&dev=true&hot=false&inlineSourceMap=true:2852:17
    at MessageQueue.__guard (index.bundle?platform=windows&dev=true&hot=false&inlineSourceMap=true:3079:13)

### Steps To Reproduce

Fresh install >
install react-native-async-storage

### Expected Results

Fetch should work

### CLI version

7.0.3

### Environment

```markdown
OS: Windows 10 10.0.22000
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Memory: 16.61 GB / 31.73 GB
  Binaries:
    Node: 16.13.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.17 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.1.3 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Enabled
      AllowAllTrustedApps: Enabled
      Versions: 10.0.18362.0, 10.0.19041.0, 10.0.22000.0
  IDEs:
    Android Studio: Not Found
    Visual Studio: 17.0.32126.317 (Visual Studio Professional 2022)
  Languages:
    Java: Not Found
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2
    react-native: 0.68.0 => 0.68.0
    react-native-windows: 0.68.4 => 0.68.4
  npmGlobalPackages:
    *react-native*: Not Found
```

### Target Platform Version

10.0.16299

### Target Device(s)

Desktop

### Visual Studio Version

Visual Studio 2022

### Build Configuration

Debug

### Snack, code example, screenshot, or link to a repository

_No response_

### PR DESCRIPTION
**Pull request by JunielKatarn**
## Description
Do not add BlobModule nor FileReaderModule to the default CxxModule list when running a UWP MSRN instance.

### Type of Change
- Bug fix (non-breaking change which fixes an issue)

### Why
Undefined object sharing behavior in UWP (as opposed to Desktop) will cause a malfunction of HTTP/fetch workflows due to a misconfigured Blob module.

Resolves #10036

### What
- Conditionally compile the code that adds `BlobModule` and `FileReaderModule` to the default modules' list in the `InstanceImpl` class.
- Defined `Blob.EnableModule` runtime option.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/10079)

### CODE REVIEW

**Comment by asklar** [comment]
> Undefined object sharing behavior in UWP (as opposed to Desktop) will cause a malfunction of HTTP/fetch workflows due to a misconfigured Blob module.
@JunielKatarn could you clarify what about UWP makes this not work / how it's related to the "invalid response" error? Ideally we don't add further behavior forking unless there's a clear platform difference

**Comment by JunielKatarn** [comment]
> > Undefined object sharing behavior in UWP (as opposed to Desktop) will cause a malfunction of HTTP/fetch workflows due to a misconfigured Blob module.
> > @JunielKatarn could you clarify what about UWP makes this not work / how it's related to the "invalid response" error? Ideally we don't add further behavior forking unless there's a clear platform difference

Updating the summary.
In short, it's the ad-hoc property bag used for the Desktop DLL which does not get used by MSRN's modules.
While we figure out a common way to initialize such property bag for both variants, the decision is not to fork functionality, but to postpone enabling Blob/FileReader in UWP.
Meaning, UWP's behavior is mean to stay unchanged for now.

**Comment by asklar** [comment]
> > > Undefined object sharing behavior in UWP (as opposed to Desktop) will cause a malfunction of HTTP/fetch workflows due to a misconfigured Blob module.
> > > @JunielKatarn
> > >             
> > >             Julio Cesar Rocha
> > >             FTE could you clarify what about UWP makes this not work / how it's related to the "invalid response" error? Ideally we don't add further behavior forking unless there's a clear platform difference
> 
> Updating the summary. In short, it's the ad-hoc property bag used for the Desktop DLL which does not get used by MSRN's modules. While we figure out a common way to initialize such property bag for both variants, the decision is not to fork functionality, but to postpone enabling Blob/FileReader in UWP. Meaning, UWP's behavior is mean to stay unchanged for now.

We have ReactInstanceSettings.Properties as the mechanism for storing/retrieving arbitrary properties in RNW
a) can Office not use that? (cc @acoates-ms for comment)
b) Is there an issue filed/ETA to undo this workaround and get us at parity? 
c) it looks like the breaking change was backported a few releases back, could you please also fix it there

I'm a little concerned that this change went through into public OSS without testing, as any basic [manual] validation with RNTester would have uncovered that this was a breaking change, and then it got backported, breaking OSS customers. I imagine this worked fine for Office Desktop but clearly not for RNW. Could we please slow down in the future and make sure we don't jam things through (esp when a change is backported) without proper validation?

**Comment by JunielKatarn** [comment]
> > > > Undefined object sharing behavior in UWP (as opposed to Desktop) will cause a malfunction of HTTP/fetch workflows due to a misconfigured Blob module.
> > > > @JunielKatarn
> > > > ```
> > > >         Julio Cesar Rocha
> > > >         FTE could you clarify what about UWP makes this not work / how it's related to the "invalid response" error? Ideally we don't add further behavior forking unless there's a clear platform difference
> > > > ```
> > 
> > 
> > Updating the summary. In short, it's the ad-hoc property bag used for the Desktop DLL which does not get used by MSRN's modules. While we figure out a common way to initialize such property bag for both variants, the decision is not to fork functionality, but to postpone enabling Blob/FileReader in UWP. Meaning, UWP's behavior is mean to stay unchanged for now.
> 
> We have ReactInstanceSettings.Properties as the mechanism for storing/retrieving arbitrary properties in RNW a) can Office not use that? (cc @acoates-ms for comment) b) Is there an issue filed/ETA to undo this workaround and get us at parity? c) it looks like the breaking change was backported a few releases back, could you please also fix it there
> 
> I'm a little concerned that this change went through into public OSS without testing, as any basic [manual] validation with RNTester would have uncovered that this was a breaking change, and then it got backported, breaking OSS customers. I imagine this worked fine for Office Desktop but clearly not for RNW. Could we please slow down in the future and make sure we don't jam things through (esp when a change is backported) without proper validation?

a) No. Office uses the Desktop DLL which can't access the whole React Context / ReactInstanceSettings mechanism.
b) No ETA, but it can be done as soon as we resolve the property bag-passing issue.
c) Awaiting for approval of this PR to begin backporting.

**Review by acoates-ms — APPROVED** [review]

**Comment by omniboyOK** [comment]
I'm having this same issue after installing 0.70 today.

**Comment by JunielKatarn** [comment]
> I'm having this same issue after installing 0.70 today.

Can you please elaborate?
After this PR was merged, we managed to enable Blob support.

What error are you getting?
A GitHub issue with specific repro steps would be much appreciated :)

**Comment by jbcullis** [comment]
Same issue here - error is "Invalid response for blob"

**Comment by tero-paananen** [comment]
I have this error now on 0.69.11. Issue did not exist on 0.69.7.

**Pull request merged**
